dces-drc-integration/src/main/java/uk/gov/justice/laa/crime/dces/integration/service/ContributionService.java

```diff
@@ -54,13 +54,24 @@ public Integer processContributionUpdate(UpdateLogContributionRequest updateLogC
     @Timed(value = "laa_dces_drc_service_process_contributions_daily_files",
             description = "Time taken to process the daily contributions files from DRC and passing this for downstream processing.")
     public boolean processDailyFiles() {
-        Map<String, CONTRIBUTIONS> successfulContributions = new HashMap<>();
-        Map<String, String> failedContributions = new HashMap<>();
-        // get all the values to process via maat call
-        List<ConcurContribEntry> contributionsList = contributionClient.getContributions("ACTIVE");
-        sendContributionsToDrc(contributionsList, successfulContributions, failedContributions);
+        int defaultNoOfRecord = feature.noOfContributionRecords();
+        List<ConcurContribEntry> contributionsList;
+        int startingId = 0;
+        boolean isSuccessful = false;
+        do {
+            Map<String, CONTRIBUTIONS> successfulContributions = new HashMap<>();
+            Map<String, String> failedContributions = new HashMap<>();
+            contributionsList = contributionClient.getContributions("ACTIVE", startingId, defaultNoOfRecord);
+            if (contributionsList != null && !contributionsList.isEmpty()) {
+                sendContributionsToDrc(contributionsList, successfulContributions, failedContributions);
+                Integer contributionFileId = updateContributionsAndCreateFile(successfulContributions, failedContributions);
+                log.info("Created contribution-file ID {}", contributionFileId);
+                startingId = contributionsList.get(contributionsList.size() - 1).getConcorContributionId();
+                isSuccessful = true;
+            }
+        } while (contributionsList != null && contributionsList.size() == defaultNoOfRecord);
 
-        return updateContributionsAndCreateFile(successfulContributions, failedContributions) != null;
+        return isSuccessful;
```

Comment:
Previously this was returning true if a non-null value was returned (which in turn was the case if successfulContributions was empty). Now it will return true even if successfulContributions is empty for all calls, which looks wrong.